src/Domain/Stager.php

```diff
@@ -84,12 +108,28 @@ private function validatePreconditions(): void
         }
     }
 
-    private function runCommand(): void
+    /**
+     * @throws \Symfony\Component\Process\Exception\RuntimeException
+     */
+    private function runCommand(): string
     {
-        $this->processFactory
+        $process = $this->processFactory
             ->create(array_merge([
                 'composer',
-                sprintf('--working-dir=%s', $this->stagingDir),
-            ], $this->command));
+                "--working-dir={$this->stagingDir}",
+            ], $this->composerCommand));
+        try {
+            $process->mustRun();
```

Comment:
It'd probably be nice, in a follow-up maybe, to take advantage of the Process component's support for asynchronous execution and output streaming, so that the user doesn't have to wait very long to get output.

Reply:
Hmm. I like that from a user experience standpoint. However, I deliberately designed `Stager::stage` to have no side effects (e.g., printing directly to the shell) and to return output as a string for its caller can deal with how it wants to. (You wouldn't want Drupal core, for example, to use the class's API directly and have to buffer the output or something.) I suppose I could inject or pass in an `OutputInterface` object so as to still give the caller control over the output while preserving the class's functional purity. What do you think, @effulgentsia?

Reply:
Actually, I don't want to pass in an `OutputInterface` object either, because just ruins my layered architecture. (A web CMS shouldn't have to know anything about a console command implementation detail.) I could return a `Process` object instead of running it myself so the calling code could do whatever it wanted to with its robust API (including passing in a callable to incrementally receive the output stream). That seems like it technically breaks my layered architecture, too, but it feels different somehow. I'll think about this a little, but I'm currently leaning toward returning a `Process` instead of running it.